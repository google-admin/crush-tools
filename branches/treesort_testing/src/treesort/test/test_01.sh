test_number=01
description="two column sort"

subtest_desc=("DD,BB"
							"3,1 using index"
							"^DD,BB"
							"^DD,^BB"
							"4,^2 -p")
subtest_opts=(
'-K DD,BB'
'-k 3,1 -p'
'-K ^DD,BB'
'-K ^DD,^BB'
'-k 4,^2 -p'
)

for subtest in `seq 0 4`; do
  expected="$test_dir/test_$test_number.$subtest.expected"
  outfile="$test_dir/test_$test_number.$subtest.actual"

  $bin ${subtest_opts[$subtest]} \
       "$test_dir/test.in" \
       > "$outfile"

  if [ $? -ne 0 ] ||
     [ "`diff -q $outfile $expected`" ]; then
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" FAIL
  else
    test_status $test_number $subtest \
                "$description (${subtest_desc[$subtest]})" PASS
    rm "$outfile"
  fi
done
