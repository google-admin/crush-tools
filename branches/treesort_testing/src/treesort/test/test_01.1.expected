CCþAAþBBþDD
2009-10-01þ278þtHGNJyQcu1þ32.4092
2009-10-01þ468þA-cSNvcqnzþ28.4036
2009-10-01þ740þ7YaqxPKpCRþ17.0268
2009-10-01þ843þZfcHogunHrþ28.4036
2009-10-02þ222þ49Uj8LVH5jþ13.5003
2009-10-02þ843þ3XbaqX1keSþ32.4092
2009-10-04þ468þvOXGmYLegsþ29.5253
2009-10-04þ843þW2OIToxqAMþ17.0268
2009-10-10þ918þ6ff-bhImN7þ26.6454
2009-10-14þ279þafUGgY6q4jþ20.7444
2009-10-14þ577þarzhFJlrMBþ28.4036
2009-10-14þ965þxVciPqOBiQþ17.0268
2009-10-15þ279þWDEGX5smiWþ20.7444
2009-10-15þ577þ5TcvIxjquUþ32.4092
2009-10-15þ740þOcNUM48IS6þ22.8364
2009-10-23þ991þENItEY_hIBþ14.9256
2009-10-25þ577þ4j4n56w__lþ14.3872
2009-10-26þ279þ0dGw4UOK0Pþ17.0268
2009-10-28þ222þOCS81RVo5jþ32.4092
2009-10-28þ918þz3cqeuch4gþ20.7444
